src/main/kotlin/gg/skytils/skytilsmod/features/impl/misc/PartyAddons.kt

```diff
@@ -0,0 +1,228 @@
+/*
+ * Skytils - Hypixel Skyblock Quality of Life Mod
+ * Copyright (C) 2023 Skytils
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published
+ * by the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+package gg.skytils.skytilsmod.features.impl.misc
+
+import gg.essential.universal.utils.MCClickEventAction
+import gg.essential.universal.wrappers.message.UMessage
+import gg.essential.universal.wrappers.message.UTextComponent
+import gg.skytils.skytilsmod.Skytils
+import gg.skytils.skytilsmod.events.impl.SendChatMessageEvent
+import gg.skytils.skytilsmod.utils.*
+import net.minecraftforge.client.event.ClientChatReceivedEvent
+import net.minecraftforge.fml.common.eventhandler.SubscribeEvent
+
+/**
+ * Inspired by https://www.chattriggers.com/modules/v/HypixelUtilities
+ */
+object PartyAddons {
+    private val partyStartPattern = Regex("^§6Party Members \\((\\d+)\\)§r$")
+    private val leaderPattern = Regex("^§eParty Leader: (?<rank>.+? |§7)?(?<name>\\w+) §r(?<status>§a|§c)●§r")
+    private val playerPattern = Regex("(?<rank>.+? |§7)?(?<name>\\w+)§r(?<status>§a|§c) ● ")
+    private val party = mutableListOf<PartyMember>()
+
+    //0 = not awaiting, 1 = awaiting 2nd delimiter, 2 = awaiting 1st delimiter
+    private var awaitingMessage = 0
+
+    @SubscribeEvent
+    fun onCommandRun(event: SendChatMessageEvent) {
+        if (!Utils.inSkyblock || !Skytils.config.partyAddons) return
+        if (Utils.equalsOneOf(event.message, "/pl", "/party list", "/p list", "/party l")) {
+            awaitingMessage = 2
+        }
+    }
+
+    @SubscribeEvent
+    fun onChat(event: ClientChatReceivedEvent) {
+        if (!Utils.inSkyblock || event.type == 2.toByte() || !Skytils.config.partyAddons) return
+        val message = event.message.formattedText
+
+        if (message == "§f§r" && awaitingMessage != 0) {
+            event.isCanceled = true
+        } else if (message.startsWith("§6Party Members")) {
+            val partyStart = partyStartPattern.find(message)
```

Comment:
if you don't care about the groups do like containsMatchIn or matches

Reply:
yea true am kinda dumb ngl